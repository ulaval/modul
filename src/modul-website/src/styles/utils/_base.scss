html,
body,
.modul-body {
    background: $modul-color--body;
    height: 100%;
}
