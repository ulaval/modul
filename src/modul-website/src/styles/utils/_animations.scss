.modul-u--light-template-transition {
    &-enter-active,
    &-leave-active {
        @media (min-width: #{$m-mq--min--s}) {
            transition: transform $m-transition-duration--l ease, opacity $m-transition-duration--l ease;
        }
    }

    &-enter,
    &-leave-to {
        @media (min-width: #{$m-mq--min--s}) {
            transform: translate3d(0, 44px, 0);
            opacity: 0.6;
        }
    }
}
