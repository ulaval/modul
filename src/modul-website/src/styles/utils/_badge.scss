.modul-logo__beta {
    display: inline-flex;
    align-items: center;
    height: 12px; // Magic number
    margin-left: $m-margin--s;
    text-transform: uppercase;
    font-size: 8px; // Mafic number
    font-weight: $m-font-weight--black;
    color: $modul-color--black-grey-dark;
    background: $m-color--ul-yellow;
    padding: 2px 3px;
}

.modul-logo__new {
    display: inline-flex;
    align-items: center;
    height: 12px; // Magic number
    margin-left: $m-margin--s;
    text-transform: uppercase;
    font-size: 8px; // Mafic number
    font-weight: $m-font-weight--black;
    color: $modul-color--black-grey-dark;
    background: $m-color--hint;
    padding: 2px 3px;
}
