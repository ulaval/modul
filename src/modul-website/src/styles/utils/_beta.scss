.modul-logo__beta {
    display: inline-flex;
    align-items: center;
    height: 12px; // Magic number
    margin-left: $m-margin--s;
    text-transform: uppercase;
    font-size: 8px; // Mafic number
    font-weight: $m-font-weight--black;
    color: $modul-color--black-grey-dark;
    background: $m-color--ul-yellow;
    padding: 2px 3px;
}
