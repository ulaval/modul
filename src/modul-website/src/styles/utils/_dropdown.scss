.modul-u--dropdown-light.modul-u--dropdown-light,
.modul-u--dropdown-dark.modul-u--dropdown-dark {
    .m-dropdown__input.m-dropdown__input,
    .m-input-style__body.m-input-style__body {
        padding-right: $m-padding--s;

        @include modul-page-title();
    }

    .m-input-style {
        &::before {
            border-width: 2px;

            @media (min-width: $m-mq--min--s) {
                border-width: 3px;
            }
        }

        &::after {
            @media (min-width: $m-mq--min--s) {
                height: 3px;
            }
        }
    }
}

.modul-u--dropdown-light.modul-u--dropdown-light {
    .m-input-style {
        &__body {
            input {
                color: $modul-color--black-grey-dark;
            }
        }

        &::before {
            border-color: $modul-color--black-grey-dark !important;
        }
    }

    &:not(.m--is-open) {
        .m-dropdown__arrow {
            .m-icon-button.m-icon-button {
                color: $modul-color--black-grey-dark !important;
            }
        }
    }
}

.modul-u--dropdown-dark.modul-u--dropdown-dark {
    .m-input-style {
        &__body {
            input {
                color: #fff;
            }
        }

        &::before {
            border-color: $m-color--ul-yellow !important;
        }
    }

    &:not(.m--is-open) {
        .m-dropdown__arrow {
            .m-icon-button.m-icon-button {
                color: $m-color--ul-yellow !important;
            }
        }
    }
}
