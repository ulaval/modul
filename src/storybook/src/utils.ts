export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
