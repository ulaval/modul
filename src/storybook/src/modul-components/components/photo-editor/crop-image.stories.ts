import { storiesOf } from '@storybook/vue';
import { CROP_IMAGE_NAME } from '@ulaval/modul-components/dist/components/photo-editor/component-names';
import CropImagePlugin from '@ulaval/modul-components/dist/components/photo-editor/crop-image/crop-image';
import Vue from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

Vue.use(CropImagePlugin);

storiesOf(`${modulComponentsHierarchyRootSeparator}${CROP_IMAGE_NAME}`, module)

    .add('default', () => ({
        template: '<m-crop-image :image={} :open="true"></m-crop-image>'
    }));
