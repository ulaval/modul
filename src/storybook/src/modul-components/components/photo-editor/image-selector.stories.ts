import { storiesOf } from '@storybook/vue';
import { IMAGE_SELECTOR_NAME } from '@ulaval/modul-components/dist/components/photo-editor/component-names';
import ImageSelectorPlugin from '@ulaval/modul-components/dist/components/photo-editor/image-selector/image-selector';
import Vue from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

Vue.use(ImageSelectorPlugin);

storiesOf(`${modulComponentsHierarchyRootSeparator}${IMAGE_SELECTOR_NAME}`, module)

    .add('With a photo', () => ({
        template: '<m-image-selector url-photo="http://placekitten.com/g/192/192" :open="true"></m-image-selector>'
    }))
    .add('Without a photo', () => ({
        template: '<m-image-selector :open="true"></m-image-selector>'
    }));
