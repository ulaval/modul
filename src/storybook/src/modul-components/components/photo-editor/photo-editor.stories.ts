import { PHOTO_EDITOR_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MPhotoEditor } from '@ulaval/modul-components/dist/components/photo-editor/photo-editor';
import Vue from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${PHOTO_EDITOR_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    components: { MPhotoEditor },
    template: '<m-photo-editor :open="true" @save-image="saveImage" @delete="deleteImage"></m-photo-editor>',
    methods: {
        saveImage: () => {
            Vue.prototype.$log.log('image-saved');
        },
        deleteImage: () => {
            Vue.prototype.$log.log('image-deleted');
        }
    }
});

defaultStory.story = {
    name: 'default'
};

export const withAPhoto = () => ({
    components: { MPhotoEditor },
    template: `<m-photo-editor url-photo="http://placekitten.com/g/192/192" :open="true" @save-image="saveImage" @delete="deleteImage"></m-photo-editor>`,
    methods: {
        saveImage: () => {
            Vue.prototype.$log.log('image-saved');
        },
        deleteImage: () => {
            Vue.prototype.$log.log('image-deleted');
        }
    }
});

