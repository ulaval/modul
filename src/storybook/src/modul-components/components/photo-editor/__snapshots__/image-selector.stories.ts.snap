// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-image-selector With a photo 1`] = `
<div
  class="m-modal m-image-selector"
>
   
  <div
    class="v-portal"
    style="display: none;"
  />
</div>
`;

exports[`Storyshots modul-components|m-image-selector Without a photo 1`] = `
<div
  class="m-modal m-image-selector"
>
   
  <div
    class="v-portal"
    style="display: none;"
  />
</div>
`;
